### PR TITLE
Ensure custom handlers are called on container clicks that trigger togglePlay or restart

### PR DESCRIPTION
**Pull request by gurupras**
### Link to related issue (if applicable)
This PR fixes issue #858 --- Custom listeners were not being called when the plyr container was clicked (triggering play/pause).

### Summary of proposed changes
  - Replaced calls to `player.restart()` and `player.togglePlay()` with `proxy(...)`

### Checklist
- [x] Use `develop` as the base branch
- [x] Exclude the gulp build (`/dist` changes) from the PR
- [ ] Test on [supported browsers](https://github.com/sampotts/plyr#browser-support)

I don't know how to test on _all_ the supported browsers. So far, I have tested it on Chrome and Firefox and it works.

### CODE REVIEW

File: src/js/listeners.js [hunk 1]
```diff
@@ -386,10 +386,10 @@ class Listeners {
                 }
 
                 if (player.ended) {
-                    player.restart();
-                    player.play();
+                    this.proxy(event, player.restart, 'restart');
+                    this.proxy(event, player.togglePlay, 'play');
```
Comment (sampotts) [inline src/js/listeners.js]: Shouldn't this one be `player.play`?

Comment (gurupras) [reply src/js/listeners.js]: You're right. Fixed and pushed

**Comment by sampotts** [comment]
Changed base to develop (as per all PRs other than docs) and merged 👍

**Comment by sampotts** [comment]
Changed base to develop (as per all PRs other than docs) and merged 👍

**Pull request merged**